core/src/main/java/io/grpc/Metadata.java

```diff
@@ -642,9 +645,9 @@ private static String validateName(String n) {
       return n;
     }
 
-    private Key(String name) {
+    private Key(String name, boolean pseudo) {
```

Comment:
We have told one user in particular they can use `:status` in a supported state. It isn't _stable_ but they have a very legitimate use-case for when the RPC never gets started, needing to check the HTTP status. We'll need to figure out an alternative for them and then migrate them.

I see two alternative ways of supporting the :status use-case: an attribute on `ClientCall`, or just change the name to a `grpc-`-prefixed name like `grpc-http-status-code`. Attribute seems cleaner, but using a different name is probably easier.